doc/js_tutorials/js_setup/js_setup/js_setup.markdown

```diff
@@ -103,3 +103,76 @@ Building OpenCV.js from Source
 
     @note
     It requires `node` installed in your development environment.
+
+-#  [optional] To build `opencv.js` with threads optimization, append `--threads` option.
+
+    For example:
+    @code{.bash}
+    python ./platforms/js/build_js.py build_js --build_wasm --threads
+    @endcode
+
+    The default threads number is the logic core number of your device. You can use `cv.parallel_pthreads_set_threads_num(number)` to set threads number by yourself and use `cv.parallel_pthreads_get_threads_num()` to get the current threads number.
+
+    @note
+    You should build wasm version of `opencv.js` if you want to enable this optimization. 
+
+-#  [optional] To build `opencv.js` with wasm simd optimization, append `--simd` option.
+    To build wasm intrinsics tests, append `--build_wasm_intrin_test` option.
+
+    For example:
+    @code{.bash}
+    python ./platforms/js/build_js.py build_js --build_wasm --simd --build_wasm_intrin_test
+    @endcode
+
+    For wasm intrinsics tests, you can use the following function to test all the cases:
+    @code{.js}
+    cv.test_hal_intrin_all()
+    @endcode
+
+    And the failed cases will be print on the console. 
+    
+    If you only want to test single data type of wasm intrinsics, you can use the following functions: 
+    @code{.js}
+    cv.test_hal_intrin_uint8()
+    cv.test_hal_intrin_int8()
+    cv.test_hal_intrin_uint16()
+    cv.test_hal_intrin_int16()
+    cv.test_hal_intrin_uint32()
+    cv.test_hal_intrin_int32()
+    cv.test_hal_intrin_uint64()
+    cv.test_hal_intrin_int64()
+    cv.test_hal_intrin_float32()
+    cv.test_hal_intrin_float64()
+    @endcode
+
+    @note
+    Now only emscripten LLVM upstream backend support wasm simd, so you need to setup upstream backend environment with the following command first: 
```

Comment:
"support" -> "supports", it's better to have a link to https://emscripten.org/docs/porting/simd.html